fixtures/topology/elastic.yaml

```diff
@@ -0,0 +1,185 @@
+spec:
+  type: ElasticSearch
+  icon: elastic
+  schedule: "@every 10m"
+  id:
+    javascript: properties.id
+  components:
+    - name: nodes
+      type: ElasticSearchNode
+      properties:
+        - name: disk-space-total
+          http:
+            - endpoint: http://elasticsearch-http.default.svc:9200/_clusters/stats
+              transform:
+                include: nodes.os.fs.available
+              authentication:
+                username:
+                  value: elastic
+                password:
+                  valueFrom:
+                    secretKeyRef:
+                      name: elasticsearch-elastic-user
+        - name: disk-space-used
+          http:
+            - endpoint: http://elasticsearch-http.default.svc:9200/_clusters/stats
+              transform:
+                include: nodes.os.fs.total
+              authentication:
+                username:
+                  value: elastic
+                password:
+                  valueFrom:
+                    secretKeyRef:
+                      name: elasticsearch-elastic-user
+        - name: version
+          http:
+            - endpoint: http://elasticsearch-http.default.svc:9200/_clusters/stats
+              transform:
+                include: nodes.versions
+              authentication:
+                username:
+                  value: elastic
+                password:
+                  valueFrom:
+                    secretKeyRef:
+                      name: elasticsearch-elastic-user
+        - name: jvm-usage
+          http:
+            - endpoint: http://elasticsearch-http.default.svc:9200/_nodes/jvm
+              authentication:
+                username:
+                  value: elastic
+                password:
+                  valueFrom:
+                    secretKeyRef:
+                      name: elasticsearch-elastic-user
+        - name: memory
+          http:
+            - endpoint: http://elasticsearch-http.default.svc:9200/_clusters/stats
+              transform:
+                include: nodes.os.mem.total
+              authentication:
+                username:
+                  value: elastic
+                password:
+                  valueFrom:
+                    secretKeyRef:
+                      name: elasticsearch-elastic-user
+        - name: cores
+          http:
+            - endpoint: http://elasticsearch-http.default.svc:9200/_nodes/os
+              transform:
+                include: os.available_processors
+              authentication:
+                username:
+                  value: elastic
+                password:
+                  valueFrom:
+                    secretKeyRef:
+                      name: elasticsearch-elastic-user
+        - name: load
+    - name: index
+      type: ElasticSearchIndex
+      properties:
+        - name: name
+        - name: documents
+          http:
+            - endpoint: http://elasticsearch-http.default.svc:9200/_stats
```

Comment:
We can hit the endpoint once, and then using javascript return a list of properties

Reply:
How do I achieve that?

Reply:
Have a look at https://github.com/flanksource/canary-checker/blob/master/templating/js/k8s.js#L176